Firestore/core/src/firebase/firestore/local/local_documents_view.mm

```diff
@@ -215,6 +217,38 @@
   return results;
 }
 
+// It is possible that a `PatchMutation` can make a document match a query, even
+// if the version in the `RemoteDocumentCache` is not a match yet (waiting for
+// server to ack). To handle this, we find all document keys affected by the
+// `PatchMutation`s that are not in `existingDocs` yet, and back fill them via
+// `remoteDocumentCache.getAll`, otherwise those `PatchMutation`s will be
+// ignored because no base document can be found, and lead to missing results
+// for the query.
+DocumentMap LocalDocumentsView::AddMissingBaseDocument(
+    const std::vector<FSTMutationBatch*>& matchingBatches,
+    const DocumentMap& existingDocs) {
+  DocumentKeySet missingDocKeys;
+  for (FSTMutationBatch* batch : matchingBatches) {
+    for (FSTMutation* mutation : [batch mutations]) {
+      if ([mutation isKindOfClass:[FSTPatchMutation class]] &&
+          existingDocs.underlying_map().find([mutation key]) ==
+              existingDocs.underlying_map().end()) {
+        missingDocKeys = missingDocKeys.insert([mutation key]);
+      }
+    }
+  }
+
+  DocumentMap mergedDocs = existingDocs;
```

Comment:
In C++, this copy isn't entirely free. If you follow the suggestion above to make `existing_docs` a value, you will no longer need this copy, and I'd suggest you omit it.

Reply:
Done.